lib/datastore.dart

```diff
@@ -362,6 +364,23 @@ abstract class Transaction {}
 /// It can be used to insert/update/delete [Entity]s, lookup/query [Entity]s
 /// and allocate IDs from the auto ID allocation policy.
 abstract class Datastore {
+  /// List of required OAuth2 scopes for Datastore operation.
+  static const Scopes = DatastoreImpl.SCOPES;
+
+  /// Access Datastore using an authenticated client.
+  ///
+  /// The [client] is an authenticated HTTP client. This client must
+  /// provide access to at least the scopes in `Datastore.Scopes`.
+  ///
+  /// The [project] is the name of the Google Cloud project.
+  ///
+  /// Returs an object providing access to Datastore. The passed-in [client]
+  /// will not be closed automatically. The caller is responsible for closing
+  /// it.
+  factory Datastore(http.Client client, String project) {
+    return DatastoreImpl(client, project);
+  }
```

Comment:
@kevmoo am I missing something or did we not expose a method to create instance of `Datastore` before?

And by implication the constructor for `DatastoreDB` is useless..

Hence, using `db.dart` or `datastore.dart` is only possible if implementing a custom `Datastore` like `package:appengine` does (it does this using grpc for improved performance).

Reply:
It's abstract. Did you dig to see where the concrete impl was returned?

Reply:
Yeah, sorry, I meant I don't think an instance of `DatastoreImpl` can be created through the public API as is. I don't see any factory constructor / static function that creates an instance of `DatastoreImpl`.

Thus, the only way to get something implementing the `Datastore` interface is to use `package:appengine` (which has a grpc based implementation) or roll your own. Which seems weird, since `DatastoreImpl` is an implementation based on `package:googleapis`, and we have tests for it -- why wouldn't we expose means to create an instance of `DatastoreImpl`? (I'm guessing this an embarrassing omission :see_no_evil:)

Reply:
**Context** from `README.md`, which I'm guessing we should also fix :D

![AmKFyRHP3LjhbXC](https://user-images.githubusercontent.com/149732/126318911-e99e413a-f83b-41a1-92bd-fd9985f8b9d9.png)

Reply:
👍